chore: correctly negotiation content-type for Client Identifier document

ThisIsMissEm opened the pull request:
## Description

This brings pod-browser inline with the specification. We will advertise as `application/ld+json` if either: the Accepts header allows `application/ld+json` or if the Accepts header indicates some other random content type (e.g., text/html, then we serve as json, because there is no HTML representation, but it is handy to be able to view a client identifier document in your browser)

## Changes

- Removes nock from the API route tests
- Switches to using correct request injection testing of the handler
- Changes the endpoint to return a 400 bad request if the request method is NOT a GET request (there's no reason a service should do anything else)
- Supports sending back `application/ld+json` content-type if requested via Accept header — the default *has changed* from `application/json` to `application/ld+json` as this is still compatible with deployments of ESS 2.0.

## Testing

```npm run test -- pages/api/app.test.js```

Alternatively, try logging in in the preview environment with both ESS PodSpaces and use.id (`idp.use.id`)

## Commit checklist

- [x] All acceptance criteria are met.
- [x] Includes tests to ensure functionality and prevent regressions.
- [ ] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.

## Interested parties

@chelseapinka @tomhgmns (external, use.id) @solid-akb

ThisIsMissEm committed:
chore: correctly negotiation content-type for Client Identifier document

ThisIsMissEm commented:
Testing results:
- ESS PodSpaces ✅ 
- ESS Dev-Next ✅ 
- NSS (inrupt.net) ✅  — it doesn't verify the client identifier document
- CSS (solidcommunity.net) ✅  — iirc, this is using CSS, but it doesn't verify the client identifier document
- use.id (idp.use.id) ❌  — appears to not send an Accept header, meaning it receives `application/json` when it wants `application/ld+json`

NSeydoux commented on pages/api/app.js:
```diff
@@ -37,7 +38,33 @@ function buildAppProfile(hostname, clientId) {
 
 export default function handler(req, res) {
+  if (req.method !== "GET") {
+    return res.status(400).send("Bad Request");
```
```suggestion
    return res.status(405).send("Method not allowed");
```

NSeydoux commented on pages/api/app.test.js:
```diff
@@ -19,52 +19,148 @@
-    res.status = jest.fn(function () {
-      return this;
+  describe("errors", () => {
+    it("responds with 400 Bad Request if the request method is not GET", async () => {
```
See my suggestion to respond with 405

NSeydoux commented on pages/api/app.js:
```diff
@@ -37,7 +38,33 @@ function buildAppProfile(hostname, clientId) {
   const hostname = `https://${req.headers.host}/`;
-  res.status(200).json(buildAppProfile(hostname, clientId));
+
+  const acceptedType = accepts(req).types([
+    "application/json",
```
It isn't obvious that the first type here is returned in the absence of an `Accept` header (which I find a bit odd, as opposed to returning `false` like it does when the negotiation fails). I think it would be worth documenting in a comment.

ThisIsMissEm committed:
chore: update browserlist package for latest versions

ThisIsMissEm committed:
refactor: move logic out of constants and into a dedicated file

This is needed to keep the code coverage high for the client identifier document, and just makes sense when you think about it: a dynamic function is not a constant

ThisIsMissEm committed:
chore: improve client identifier document content-negotiation

ThisIsMissEm committed:
chore: add additional optional fields to client identifier document for compatibility with use.id

ThisIsMissEm commented:
@NSeydoux ended up also moving some logic from the constants directory to a separate file, as the api/app unit tests were complaining a bit about code coverage of the logic imported by the constants which relied on data in the window/DOM.

ThisIsMissEm commented:
Testing results:

- ESS PodSpaces ✅
- ESS Dev-Next ❌  — currently having issues due to a bad upgrade
- NSS (inrupt.net) ✅ — it doesn't verify the client identifier document
- CSS (solidweb.me) ✅
- use.id (idp.use.id) ✅  — fixed, added response_types of code, even though this is a default, use.id [doesn't respect that](https://gitlab.com/vincenttunru/penny/-/issues/25#note_1242624516), which is their bug.

ThisIsMissEm commented:
Testing results:

ESS PodSpaces ✅
ESS Dev-Next ✅ 
NSS (inrupt.net) ✅ — it doesn't verify the client identifier document
CSS (solidweb.me) ✅
use.id (idp.use.id) ✅ — fixed, added response_types of code, even though this is a default, use.id [doesn't respect that](https://gitlab.com/vincenttunru/penny/-/issues/25#note_1242624516), which is their bug.

ThisIsMissEm committed:
sonar: exclude .jsx the same as .js
